docs: add a short description about the project roadmap views (#12)
bf2fc6cc711aee1a0c2a/ffm-project: roadmap.md

# Usage of Epics
An Epic is a grouping of smaller related issues/tasks that fits into a larger goal. An Epic can potentially span across multiple iterations. 

To track how smaller issues fit into the larger goal, we use [task lists](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-task-lists).
Breaking a large issue into smaller issues makes the work more manageable and enables team members to work in parallel. 
It also leads to smaller pull requests, which are easier to review.

>NOTE: To create an epic on Github, use the available Epic template.

# Usage of Github issue template/forms
We have [customized templates](https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/blob/main/.github/ISSUE_TEMPLATE/epic.md) that helps contributors in 
creating new issues in the repository. When creating an issue, choose the template that best fits the issue you are about to create.  

# Communicate

> - Use `@mentions` to alert a person or entire team about a the issue. 

Reviewer: ```suggestion
- Use `@mentions` to alert a person or entire team about the issue. 
```

> - Use `@mentions` to alert a person or entire team about a the issue. 

Reviewer: Should we have a triage process for different types of issues? How would a creator know which is the correct person to alert about a new issue or Epic?

Author: I was hoping we could use teams for this. And then we'll have a specific team that part of the triaging process. Once that happens we'll change the `@mentions` to `@ffm/triage`

Reviewer: That makes a ton of sense. Let's tease that out now as well.

Since ffm will be used by multiple teams e.g. Quarkus and Golang fleet managers. Should there be a single triage team or separate based on the related project?

Author: > I was hoping we could use teams for this. And then we'll have a specific team that part of the triaging process. Once that happens we'll change the @mentions to @ffm/triage

Will it make sense to use the [factorized-fleet](https://github.com/bf2fc6cc711aee1a0c2a/org-management/blob/main/org/config.yaml#L947) team for now and change `@mentions` with that team?

Author: Thanks.

- Use appropriate labels during the creation or the triaging of the issue. Labels help to categorise issues thus improving their discoverbility. Check out the existing labels on - https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/labels.  
- Assign collaborators to issues to communicate responsibility.

>NOTE: You can always reach out to the wider team via the available [communication channels](https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/wiki/How-to-communicate-in-the-project) 

# View the Roadmap
The roadmap is defined in the [project view](https://github.com/orgs/bf2fc6cc711aee1a0c2a/projects/5) 

> We have two main views:

Reviewer: Do the current views make sense? I feel like we should clean these views up in the project and have clear goals on their purpose. 

Please feel free to edit the existing views and remove ones that don't make sense.

Author: I didn't change the current views. I wanted to see if there is agreement in having only two views. My reasoning are explained in https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/pull/12#discussion_r811855613

Once we have an agreement, we'll proceed with the cleaning.

Reviewer: I missed that, what you have identified makes perfect sense to me

> - The roject view containing all the issues that have been triaged and potentially assigned to a Sprint.

Reviewer: ```suggestion
- The project view contains all the issues that have been triaged and potentially assigned to a Sprint.
```

> - The Epic view, containing larger goals/tasks.

Reviewer: ```suggestion
- The Epic view, contains larger goals/tasks.
```

The two views are the source of truth and they are kept up to date to showing our roadmap.

## How to add an issue to the Roadmap (TODO)

## How to add an Epic to the Roadmap (TODO)

> # Usage of Milestone

Reviewer: I would much prefer to use the `Sprint` field in the project view to assign to a sprint rather than the milestone.

This is a custom iteration type field created in the project view and can only be assigned in the project view.

Author: Yes, the milestone are aligned to a quarterly "release". An issue could be assigned to a milestone and a sprint (using the iteration field) at the same time. My thinking is, an issue may as well be completed within a sprint which in turn is part of an even bigger / wider milestone.

Reviewer: Perfect, that sounds good to me. I am not sold on the idea of tying milestones to a quarter but the text is good.

Can you add a section on Sprints and using that field as well, please. Even if it is to say that the sprint field should generally be assigned during sprint planning.

Author: > Perfect, that sounds good to me. I am not sold on the idea of tying milestones to a quarter but the text is good.

I think how we define milestones will be tied to our release process. If we say that we'll be doing quarterly release (creating a new version), that means a release each 3 months, to be that sounds like less frequent. I am fine too with keeping the text as it is now and refine it as we go.  

> Can you add a section on Sprints and using that field as well, please. Even if it is to say that the sprint field should generally be assigned during sprint planning.

Good point. I'll add it. I remember having mentioned Sprint without clearly saying how it is defined.

creating new issues in the repository. When creating an issue, choose the template that best fits the issue you are about to create.  

# Communicate
- Use `@mentions` to alert a person or entire team about a the issue. 
- Use appropriate labels during the creation or the triaging of the issue. Labels help to categorise issues thus improving their discoverbility. Check out the existing labels on - https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/labels.  
- Assign collaborators to issues to communicate responsibility.

>NOTE: You can always reach out to the wider team via the available [communication channels](https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/wiki/How-to-communicate-in-the-project) 

# View the Roadmap

The roadmap is defined in the [project view](https://github.com/orgs/bf2fc6cc711aee1a0c2a/projects/5) 
We have two main views:
- The roject view containing all the issues that have been triaged and potentially assigned to a Sprint.
- The Epic view, containing larger goals/tasks.

The two views are the source of truth and they are kept up to date to showing our roadmap.

## How to add an issue to the Roadmap (TODO)

## How to add an Epic to the Roadmap (TODO)

> 

Author: I have captured David thoughts on having two views:
- the project view containing all issues assigned to an iteration. I think some issues (un-triaged ones) might be assigned to an iteration and that is fine as we can consider those issues to be the one in the backlog
- the epic view containing larger chunks of work 

This is reasonable enough as the epic view will complement the project view containing all issues. Some people might be interested in just seeing the big picture and its ETA.
The project view will contain all issues, useful enough to show the ETA of individual issues. I can also see Bugs report issue to be seen here as they do not have a parent Epic.

Reviewer: That makes sense to me @machi1990 . So the project view is ultimately the sprint board and backlog combined? The Epic view is looking at the higher level scope of work.

Author: > So the project view is ultimately the sprint board and backlog combined? 

Correct. And if someone is interested on seeing the work on a given sprint e.g current sprint, filters can be leveraged. This might be simplistic but it avoid the case of juggling through and maintaining  many views.

Author: > I have captured David thoughts on having two views:
> 
> * the project view containing all issues assigned to an iteration. I think some issues (un-triaged ones) might be assigned to an iteration and that is fine as we can consider those issues to be the one in the backlog
> * the epic view containing larger chunks of work
> 
> This is reasonable enough as the epic view will complement the project view containing all issues. Some people might be interested in just seeing the big picture and its ETA. The project view will contain all issues, useful enough to show the ETA of individual issues. I can also see Bugs report issue to be seen here as they do not have a parent Epic.

As I continue to tweak the board, I feel like we can have only one project view  containing all the issues.
Then the epic view can simply simply be obtained using the `label: kind/epic` filter. 
The fundamental reason is that, if we add all the issues to the project view (epics, issues under epic and bugs) this also ends up messing the epic view because everything will be contained there too. So one has to filter out to get the epics.

Reviewer: It is not possible to force a filter for an individual view?

Author: Unfortunately, I could not find a way to do so.

It also leads to smaller pull requests, which are easier to review.

>NOTE: To create an epic on Github, use the available Epic template.

# Usage of Github issue template/forms
We have [customized templates](https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/blob/main/.github/ISSUE_TEMPLATE/epic.md) that helps contributors in 
creating new issues in the repository. When creating an issue, choose the template that best fits the issue you are about to create.  

# Communicate
- Use `@bf2fc6cc711aee1a0c2a/factorized-fleet` to alert the team about the issue. 
- Use appropriate labels during the creation or the triaging of the issue. Labels help to categorise issues thus improving their discoverbility. Check out the existing labels on - https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/labels.  
- Assign collaborators to issues to communicate responsibility.

>NOTE: You can always reach out to the wider team via the available [communication channels](https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/wiki/How-to-communicate-in-the-project) 

# Assigning issue to a Sprint

> A Sprint is defined as an [iteration field](https://docs.github.com/en/issues/trying-out-the-new-projects-experience/best-practices-for-managing-projects#use-different-field-types). An issue is assigned to a Sprint during Sprint planning.

Reviewer: ```suggestion
A Sprint is defined as an [iteration field](https://docs.github.com/en/issues/trying-out-the-new-projects-experience/best-practices-for-managing-projects#use-different-field-types) in the GitHub project view. An issue is assigned to a Sprint during Sprint planning.
```

> # View the Roadmap

Reviewer: This would be more ideal as the first item in the document. Since the file name is also `roadmap`

The roadmap is defined in the [project view](https://github.com/orgs/bf2fc6cc711aee1a0c2a/projects/5) 
The project view contains all the issues that have been triaged and potentially assigned to a Sprint.   

The view is the source of truth and they are kept up to date to showing our roadmap.

## How to add an issue to the Roadmap

To add an issue to the Roadmap, go the [project](https://github.com/orgs/bf2fc6cc711aee1a0c2a/projects/5) and click on the `Add Item` button (displayed down the board). 
Github will propose a search input, where you can search for issues and add them to the board.

> ## How to assign an issue to the Sprint

Reviewer: This is kind of duplicated with the heading 1 section above "Assigning issue to a Sprint". I would advise merging the two.

Go to the [Project table view](https://github.com/orgs/bf2fc6cc711aee1a0c2a/projects/5/views/4?layout=table) which contains the `Sprint` column. 
Look for the issue you are interested in. Then on the Sprint column, you could select a Sprint from the drop down list and assign it to the issue.

## Filtering

On the project you could filter by Label, Sprint, assignee, Status etc. See the [available filtering options](https://docs.github.com/en/issues/organizing-your-work-with-project-boards/tracking-work-with-project-boards/filtering-cards-on-a-project-board) 

### Show only Epic

To show Epics, use the filter `label: kind/epic`. This will show only larger goals issue.

### Issue in a Sprint

To show issues that are being progressed in a Spring `X`, use the filter `sprint:X` (where `X` is Sprint value).
To show all issues in the current Sprint use the `sprint:@current` filter.

# Usage of Milestone
Milestones will be used as rough outlines of work assigned to a quarter. A milestone can be assigned to a smaller issue or an epic.
Check out the available milestones - https://github.com/bf2fc6cc711aee1a0c2a/ffm-project/milestones.

> # Automation

Reviewer: This feels out of place. I would either remove it or caveat it in the sentence to explicitly call out that we want to introduce automation.

Author: Yes, this is very much a TODO section. I am okay to remove it and it will be added when we have automation in place.
